test/src/test/java/org/corfudb/runtime/checkpoint/DistributedCompactorTest.java

```diff
@@ -0,0 +1,505 @@
+            token = (TokenMsg) txn.getRecord(DistributedCompactor.CHECKPOINT, DistributedCompactor.CHECKPOINT_KEY).getPayload();
+            txn.commit();
+        }
+
+        return token != null;
+    }
+
+    @Test
+    public void initTest() {
+        CompactorLeaderServices compactorLeaderServices1 = new CompactorLeaderServices(runtime0, SERVERS.ENDPOINT_0);
+        compactorLeaderServices1.setLeader(true);
+        CompactorLeaderServices compactorLeaderServices2 = new CompactorLeaderServices(runtime1, SERVERS.ENDPOINT_1);
+        compactorLeaderServices2.setLeader(false);
+
+        assert compactorLeaderServices1.trimAndTriggerDistributedCheckpointing() == LeaderServicesStatus.SUCCESS;
+        assert compactorLeaderServices2.trimAndTriggerDistributedCheckpointing() == LeaderServicesStatus.FAIL;
+        assert verifyManagerStatus(StatusType.STARTED);
+        assert verifyCheckpointStatusTable(StatusType.IDLE, 0);
+    }
+
+    @Test
+    public void initMultipleLeadersTest1() {
+        CompactorLeaderServices compactorLeaderServices1 = new CompactorLeaderServices(runtime0, SERVERS.ENDPOINT_0);
+        compactorLeaderServices1.setLeader(true);
+        CompactorLeaderServices compactorLeaderServices2 = new CompactorLeaderServices(runtime1, SERVERS.ENDPOINT_1);
+        compactorLeaderServices2.setLeader(true);
+
+        LeaderServicesStatus init1 = compactorLeaderServices1.trimAndTriggerDistributedCheckpointing();
+        LeaderServicesStatus init2 = compactorLeaderServices2.trimAndTriggerDistributedCheckpointing();
+
+        assert !init1.equals(init2);
+        assert verifyManagerStatus(StatusType.STARTED);
+        assert verifyCheckpointStatusTable(StatusType.IDLE, 0);
+    }
+
+    @Test
+    public void initMultipleLeadersTest2() {
+        CompactorLeaderServices compactorLeaderServices1 = new CompactorLeaderServices(runtime0, SERVERS.ENDPOINT_0);
+        compactorLeaderServices1.setLeader(true);
+        CompactorLeaderServices compactorLeaderServices2 = new CompactorLeaderServices(runtime1, SERVERS.ENDPOINT_1);
+        compactorLeaderServices2.setLeader(true);
+
+        ExecutorService scheduler = Executors.newFixedThreadPool(2);
+        Future<LeaderServicesStatus> future1 = scheduler.submit(() -> compactorLeaderServices1.trimAndTriggerDistributedCheckpointing());
+        Future<LeaderServicesStatus> future2 = scheduler.submit(() -> compactorLeaderServices2.trimAndTriggerDistributedCheckpointing());
+        compactorLeaderServices1.setLeader(false);
+
+        try {
+            if (!future1.get().equals(future2)) {
+                assert verifyManagerStatus(StatusType.STARTED);
+                assert verifyCheckpointStatusTable(StatusType.IDLE, 0);
+            } else {
+                assert future1.get() != LeaderServicesStatus.SUCCESS && future2.get() != LeaderServicesStatus.SUCCESS;
+            }
+        } catch (Exception e) {
+            log.warn("Unable to get results");
```

Comment:
Consider failing the test if this occurs since the assertions may not have been verified.